File: TheAnvil/src/main/java/com/qualityplus/anvil/base/session/AnvilSessionImpl.java
package com.qualityplus.anvil.base.session;

import com.qualityplus.anvil.api.session.AnvilSession;
import com.qualityplus.anvil.util.AnvilFinderUtil;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.bukkit.inventory.ItemStack;

@Getter
@AllArgsConstructor
public final class AnvilSessionImpl implements AnvilSession {
    private final ItemStack result;
    private final ItemStack itemToUpgrade;
    private final ItemStack itemToSacrifice;

    @Override
    public SessionResult getSessionResult() {
        return AnvilFinderUtil.getAnswer(this);
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/base/session/AnvilSessionImpl.kt
package com.qualityplus.anvil.base.session

import com.qualityplus.anvil.api.session.AnvilSession
import com.qualityplus.anvil.util.AnvilFinderUtil
import lombok.AllArgsConstructor
import lombok.Getter
import org.bukkit.inventory.ItemStack

@Getter
@AllArgsConstructor
class AnvilSessionImpl : AnvilSession {
    override var result: ItemStack? = null
    override var itemToUpgrade: ItemStack? = null
    override var itemToSacrifice: ItemStack? = null

    override val sessionResult: AnvilSession.SessionResult
        get() = AnvilFinderUtil.getAnswer(this)
}